### PR TITLE
DS-3474 Add PropensityWeights function

### PR DESCRIPTION
**Pull request by jrwishart**
- DS-3474 First draft of PropensityWeights function [revdep skip]
- DS-3474 Fix survey@4.0
- DS-3474 Use predict instead of predict_proba
- DS-3474 Update classification predictions
- DS-3474 Expect warnings from older XGBoost rds files
- DS-3474 Add dep on flipRegression
- DS-3474 Temporarily point to flipRegression dev on travis

### CODE REVIEW

File: R/variables.R [hunk 1]
```diff
@@ -250,4 +250,38 @@ Probabilities.GradientBoost <- function(object)
...
+    n.classes <- NCOL(probabilities)
+    if (n.classes > 2L)
+        stop("The supplied model is a multiclass classification model with ", n.classes, " ",
+             "outcome categories/class labels. Computing Propensity weights is only supported ",
+             "for binary classification models. E.g. Binary Logit, or a Machine Learning model ",
+             "with a Binary variable as the outcome variable. Consider merging categories ",
+             "in the outcome variable to produce a binary classification before computing ",
+             "propensity weights.")
+    outcome.variable <- OutcomeVariable(object[["formula"]], object[["model"]])
+    if (n.classes != 2L)
```
Comment (mwmclean) [inline R/variables.R:281]: Why is this needed if you've already checked if `n.classes > 2L`? How could `n.classes` be anything other than 2 given the checks on L268 and L272?

**Review by mwmclean — APPROVED** [review]

File: R/variables.R (hunk 1 above)
Comment (jrwishart) [reply R/variables.R:281]: Its a weird edge case but still possible. It can occur that a binary logit model can be created where all the outcome observations are of the same class. The user is simply supplied with a warning in that case. That probably shouldn't just warn and it should error instead. For example, 
image

Comment (jrwishart) [reply R/variables.R:281]: By same class I mean the outcome variable is a factor with only one level.
